server/api_search.go

```diff
@@ -0,0 +1,339 @@
+// Copyright (c) 2023-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"strings"
+
+	"github.com/gin-gonic/gin"
+	"github.com/mattermost/mattermost-plugin-ai/server/embeddings"
+	"github.com/mattermost/mattermost-plugin-ai/server/llm"
+	"github.com/mattermost/mattermost/server/public/model"
+)
+
+const (
+	SearchResultsProp = "search_results"
+	SearchQueryProp   = "search_query"
+)
+
+type SearchRequest struct {
+	Query      string `json:"query"`
+	TeamID     string `json:"teamId"`
+	ChannelID  string `json:"channelId"`
+	MaxResults int    `json:"maxResults"`
+}
+
+type SearchResponse struct {
+	Answer    string      `json:"answer"`
+	Results   []RAGResult `json:"results"`
+	PostID    string      `json:"postId,omitempty"`
+	ChannelID string      `json:"channelId,omitempty"`
+}
+
+type RAGResult struct {
+	PostID      string  `json:"postId"`
+	ChannelID   string  `json:"channelId"`
+	ChannelName string  `json:"channelName"`
+	UserID      string  `json:"userId"`
+	Username    string  `json:"username"`
+	Content     string  `json:"content"`
+	Score       float32 `json:"score"`
+}
+
+// convertToRAGResults converts embeddings.SearchResult to RAGResult with enriched metadata
+func (p *Plugin) convertToRAGResults(searchResults []embeddings.SearchResult) []RAGResult {
+	var ragResults []RAGResult
+	for _, result := range searchResults {
+		// Get channel name
+		var channelName string
+		channel, chErr := p.pluginAPI.Channel.Get(result.Document.ChannelID)
+		if chErr != nil {
+			p.pluginAPI.Log.Warn("Failed to get channel", "error", chErr, "channelID", result.Document.ChannelID)
+			channelName = "Unknown Channel"
+		} else {
+			if channel.Type == model.ChannelTypeDirect || channel.Type == model.ChannelTypeGroup {
+				channelName = "Direct Message"
```

Comment:
Makes sense to make any distinction between direct and group? I mean, maybe the channelName should be "Group Message" in case of that being a group message.

Reply:
Done.